metadata-service/auth-impl/src/main/java/com/datahub/authorization/PolicyEngine.java

```diff
@@ -49,37 +47,22 @@ public class PolicyEngine {
 
   public PolicyEvaluationResult evaluatePolicy(
       final DataHubPolicyInfo policy,
-      final String actorStr,
-      final String privilege,
-      final Optional<ResolvedResourceSpec> resource) {
-    try {
-      // Currently Actor must be an urn. Consider whether this contract should be pushed up.
-      final Urn actor = Urn.createFromString(actorStr);
-      return evaluatePolicy(policy, actor, privilege, resource);
-    } catch (URISyntaxException e) {
-      log.error(String.format("Failed to bind actor %s to an URN. Actors must be URNs. Denying the authorization request", actorStr));
-      return PolicyEvaluationResult.DENIED;
-    }
-  }
-
-  public PolicyEvaluationResult evaluatePolicy(
-      final DataHubPolicyInfo policy,
-      final Urn actor,
+      final ResolvedResourceSpec actorResolvedResourceSpec,
```

Comment:
Minor: I see we are reusing the concept of "ResourceSpec" here, but I'm not sure we want to. The actor is not usually thought of as a "resource", but is its own special concept. Maybe we have a ResolvedActorSpec and ActorSpec classes of their own?

Reply:
What do you think? 

Otherwise, we may be able to rename ResourceSpec as EntitySpec. 

I want to minimize the amount of confusion we introduce here

Reply:
I mean my preference would be to add a parallel set of concepts: 

Actor and ResolvedActor 

or something similar, to keep the two concepts clearly distinct.

Reply:
We have decided to rename ResourceSpec to EntitySpec (and all other related classes) to avoid the additional complexity of parallel classes with almost the same behavior.